src/rpm/builder.rs

```diff
@@ -198,28 +198,62 @@ impl RPMBuilder {
     }
 
-    /// Returns a value that can be used for associating several package builds as being part of one operation
+    /// Define a value that can be used for associating several package builds as being part of one operation
     ///
     /// You can use any value, but the standard format is "${build_host} ${build_time}"
     pub fn cookie(mut self, cookie: impl AsRef<str>) -> Self {
         self.cookie = Some(cookie.as_ref().to_owned());
         self
     }
 
-    pub fn compression(mut self, comp: CompressionType) -> Self {
-        self.compressor = comp;
+    /// Set the compression type and/or level to be used for the payload of the built package
+    ///
+    /// Passing a `CompressionType` value will use a default compression level which has been
+    /// optimized for package size over compression time.
+    ///
+    /// ```
+    /// # fn foo() -> Result<(), Box<dyn std::error::Error>> {
+    ///
+    /// let pkg = rpm::RPMBuilder::new("foo", "1.0.0", "MIT", "x86_64", "some baz package")
+    ///     .compression(rpm::CompressionType::Gzip)
+    ///     .build()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// If you would like to specify a custom compression level (for faster package builds, at the
+    /// expense of package size), pass a `CompressionDetails` value instead.
+    ///
+    /// ```
+    /// # fn foo() -> Result<(), Box<dyn std::error::Error>> {
+    ///
+    /// let pkg = rpm::RPMBuilder::new("foo", "1.0.0", "MIT", "x86_64", "some baz package")
+    ///     .compression(rpm::CompressionDetails::Zstd(3))
+    ///     .build()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// For Gzip compression, the expected range is 0 to 9, with a default value of 9.
+    /// For Xz compression, the expected range is 0 to 9, with a default value of 9.
+    /// For Zstd compression, the expected range is 1 to 22, with a default value of 19.
+    ///
+    /// If this method is not called, the payload will be Gzip compressed by default. This may change
+    /// in future versions of the library.
+    pub fn compression<T: Into<CompressionDetails>>(mut self, comp: T) -> Self {
+        self.compression = comp.into();
         self
     }
 
-    /// Add an entry to the changelog, compromised of changelog name (which includes author, email followed by
-    /// a dash followed by a version number),
-    /// description, and the date and time of the change.
+    /// Add an entry to the package changelog.
+    ///
+    /// The a changelog entry consists of an entry name (which includes author, email followed by
+    /// a dash followed by a version number), description, and the date and time of the change.
 
     /// ```
     /// # fn foo() -> Result<(), Box<dyn std::error::Error>> {
     /// use rpm::chrono::TimeZone;
     ///
-    /// let pkg = rpm::RPMBuilder::new("foo", "1.0.0", "Apache-2.0", "x86_64", "some baz package")
+    /// let pkg = rpm::RPMBuilder::new("foo", "1.0.0", "MIT", "x86_64", "some baz package")
```

Comment:
I don't understand why all examples are unified to MIT?

Reply:
It's just shorter

Reply:
Please revert